app/rules/RuleProvisionerService.scala

```diff
@@ -0,0 +1,44 @@
+package rules
+
+import java.util.Date
+
+import akka.actor.Scheduler
+import matchers.RegexMatcher
+import services.MatcherPool
+import utils.Loggable
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+class RuleProvisionerService(ruleBucket: BucketRuleResource, matcherPool: MatcherPool)(implicit ec: ExecutionContext) extends Loggable with Runnable {
+  var lastModified: Date = new Date(0)
+
+  def updateRules(): Unit = {
+    matcherPool.removeAllMatchers()
```

Comment:
Would it be best to remove our matchers once we've read from S3 and serialised our rules? We don't yet know when we can read new rules from the bucket.